### PR TITLE
Finish wiring PrintFlags

### PR DESCRIPTION
**Pull request by soltysh**
**What this PR does / why we need it**:

This is rebased version of #63055, I don't want to wait for @juanvallejo to return and to unblock myself, and him for next week. I've rebased his PR and will push it forward.

/assign @deads2k 

since you "broke" us :wink: 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/scale.go [hunk 1]
```diff
@@ -227,8 +267,14 @@ func (o *ScaleOptions) RunScale(f cmdutil.Factory, out, errOut io.Writer, cmd *c
...
-		cmdutil.PrintSuccess(shortOutput, out, info.Object, false, "scaled")
-		return nil
+		internalObj, err := legacyscheme.Scheme.ConvertToVersion(info.Object, info.Mapping.GroupVersionKind.GroupKind().WithVersion(runtime.APIVersionInternal).GroupVersion())
+		if err != nil {
+			// if there's an error, try to print what you have (mirrors old behavior).
+			glog.V(1).Info(err)
+			return o.PrintObj(info.Object, o.Out)
+
+		}
+		return o.PrintObj(internalObj, o.Out)
```
Comment (deads2k) [inline pkg/kubectl/cmd/scale.go]: this won't work.  You can't print an internal to json or yaml.

**Comment by deads2k** [comment]
@juanvallejo you'll want to see if there's a way you can detect and reject internal types.  I'll even accept super ugly, reflect and check if the package looks like a version.

File: pkg/kubectl/cmd/scale.go (hunk 1 above)
Comment (soltysh) [reply pkg/kubectl/cmd/scale.go]: Lemme check that again...

Comment (soltysh) [reply pkg/kubectl/cmd/scale.go]: It was wrong in the original PR, I've changed it to `info.AsVersioned()` as we do it in all the places.

**Pull request merged**